cms/djangoapps/contentstore/management/commands/reindex_course.py

```diff
@@ -135,16 +146,27 @@ def handle(self, *args, **options):
 
         total = len(course_keys)
         logging.warning(f'Reindexing {total} courses...')
-        reindexed = 0
         start = time()
 
+        count = 0
+        success = 0
+        errors = []
+
         for course_key in course_keys:
             try:
+                count += 1
                 CoursewareSearchIndexer.do_course_reindex(store, course_key)
-                reindexed += 1
-                if reindexed % 10 == 0 or reindexed == total:
-                    now = time()
-                    t = now - start
-                    logging.warning(f'{reindexed}/{total} reindexed in {t:.1f} seconds.')
+                success += 1
+                if count % 10 == 0 or count == total:
+                    t = time() - start
+                    remaining = total - success - len(errors)
+                    logging.warning(f'{success} courses reindexed in {t:.1f} seconds. {remaining} remaining...')
             except Exception as exc:  # lint-amnesty, pylint: disable=broad-except
+                errors.append(course_key)
                 logging.exception('Error indexing course %s due to the error: %s.', course_key, exc)
+
+        t = time() - start
+        logging.warning(f'{success} of {total} courses reindexed succesfully. Total running time: {t:.1f} seconds.')
+        if errors:
+            logging.warning('Reindex failed for %s courses:\n%s\n', len(errors),
+                            ''.join(f"{course_key}\n" for course_key in errors))
```

Comment:
I don't know if our loggers will lose it logging a 100 line long message - maybe give each one its own log line?

Reply:
Done.